src/main/java/sg/ncl/MainController.java

```diff
@@ -471,10 +477,117 @@ public String loginSubmit(
 
     }
 
-    @RequestMapping("/passwordreset")
-    public String passwordreset(Model model) {
-        model.addAttribute("loginForm", new LoginForm());
-        return "passwordreset";
+    // triggered when user clicks "Forget Password?"
+    @RequestMapping("/password_reset_email")
+    public String passwordResetEmail(Model model) {
+        model.addAttribute("passwordResetRequestForm", new PasswordResetRequestForm());
+        return "password_reset_email";
+    }
+
+    // triggered when user clicks "Send Reset Email" button
+    @PostMapping("/password_reset_request")
+    public String sendPasswordResetRequest(
+            @ModelAttribute("passwordResetRequestForm") PasswordResetRequestForm passwordResetRequestForm
+    ) throws WebServiceRuntimeException
+    {
+        String email = passwordResetRequestForm.getEmail();
+        if(!VALID_EMAIL_ADDRESS_REGEX.matcher(email).matches()) {
+            passwordResetRequestForm.setErrMsg("Please provide a valid email address");
+            return "password_reset_email";
+        }
+
+        JSONObject obj = new JSONObject();
+        obj.put("username", email);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> request = new HttpEntity<>(obj.toString(), headers);
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = null;
+        try {
+            response = restTemplate.exchange(properties.getPasswordResetRequestURI(), HttpMethod.POST, request, String.class);
+        } catch (RestClientException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)